src/patternfly/components/ModalBox/styles.scss

```diff
@@ -41,6 +42,7 @@
   min-height: var(--pf-c-modal-box--MinHeight);
   max-height: var(--pf-c-modal-box--MaxHeight);
   background-color: var(--pf-c-modal-box--BackgroundColor);
+  border: var(--pf-c-modal-box--BorderSize) solid transparent;
```

Comment:
Where you have `transparent` here, should be replaced with a variable named `--pf-c-modal-box-BorderColor`, and this variable should be added to the top like this:
`--pf-c-modal-box-BorderColor: transparent;`

This will let consumers retheme the modal box to have a visible border.